feat: disable html5 validations by default (#2307)
rjsf-team/react-jsonschema-form: docs/usage/validation.md

## HTML5 Validation

> By default, the form does not use HTML5 validation. 

Reviewer: ```suggestion
By default, the form does not use [HTML5's built-in validation](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation). Rather, react-jsonschema-form's validation is used by default. 
```

> If you wish to enable HTML5 validation, passing `html5Validate=false` will enable it. 

Reviewer: ```suggestion
If you wish to enable HTML5 validation, passing `html5Validate=true` will enable it. 
```

> If you wish to enable HTML5 validation, passing `noHtml5Validate=false` will enable it. 

Reviewer: Do you think this is intuitive enough (seems kind of strange that someone would need to pass in "false" to enable a feature)? Would it make better sense to create a new prop called `html5Validate=true`, or do you think `noHtml5Validate=false` is clear enough?

Author: I was mostly keeping the props consistent, but if you're cool with swapping that out, I think that makes more sense.   I'll udpate the code/tests/docs to do so

Reviewer: Yeah, I guess I'm not sure. I'm cool with swapping it out if it makes sense, do you think it makes sense to do so?

> You may find that enabling HTML5 validation errors leads to some unintuitive results, as this will display some errors prior to form submission in a manner different than the way errors from react-jsonschema-form displays validation errors

Reviewer: ```suggestion
You may find that enabling HTML5 validation errors leads to some unintuitive results, as this will display some errors prior to form submission in a manner different than the way that react-jsonschema-form displays validation errors.
```

};

render((

>   <Form schema={schema} html5Validate={false} />

Reviewer: ```suggestion
  <Form schema={schema} html5Validate={true} />
```
